source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst

```diff
@@ -0,0 +1,95 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_set_matrix_property:
+
+set_matrix_property
+===================
+
+Matrix properties
+-----------------
+
+.. container:: section
+
+   .. code:: cpp
+
+      enum class matrix_property {
+          symmetric,
+          sorted,
+      };
+
+   TODO: Clarify are the properties meant to be used as a bit mask so multiple
+   properties can be set?
```

Comment:
@gajanan-choudhary were you planning to have properties that can combine with each other, for instance if a matrix is both symmetric and sorted? If so we could consider bit masks.

Reply:
Each property should be considered orthogonal to other properties.  Some dispatchers internally may use multiple properties to select a codepath, but each one should be individually or collectively settable/readable.  bitmasks could be fine for implementing this.

Reply:
This is done https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/dd541fe3c69a52bec136cdb9aaa933289047b80d